Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs

```diff
@@ -706,39 +706,99 @@ string stdInClientHandle
             SendEvent(new AP.ModulesChangedEvent());
         }
 
-        /// <summary>
-        /// Creates a new ProjectEntry for the collection of buffers.
-        /// </summary>
-        internal async void ReAnalyzeTextBuffers(BufferParser oldParser) {
-            ITextBuffer[] buffers = oldParser.Buffers;
-            if (buffers.Length > 0) {
-                if (!oldParser.AnalysisEntry.SuppressErrorList) {
-                    _services.ErrorTaskProvider?.ClearErrorSource(oldParser.AnalysisEntry, ParserTaskMoniker);
-                    _services.ErrorTaskProvider?.ClearErrorSource(oldParser.AnalysisEntry, UnresolvedImportMoniker);
-                    _services.CommentTaskProvider?.ClearErrorSource(oldParser.AnalysisEntry, ParserTaskMoniker);
+        internal void ReanalyzeEntry(AnalysisEntry entry) {
+            var parser = entry.TryGetBufferParser();
+            if (parser == null) {
+                return;
+            }
+            var buffers = parser.Buffers;
+            if (!buffers.Any()) {
+                return;
+            }
+
+            foreach (var buffer in buffers) {
+                parser.RemoveBuffer(buffer);
+            }
+            
+        }
+
+        internal async Task TransferFromOldAnalyzer(VsProjectAnalyzer oldAnalyzer) {
+            var oldFileAndEntry = oldAnalyzer.LoadedFiles.ToArray();
+            var oldFiles = oldFileAndEntry.Select(kv => kv.Key);
+            var oldEntries = oldFileAndEntry.Select(kv => kv.Value);
+
+            var oldBuffers = oldEntries.ToDictionary(e => e.Path, e => e.TryGetBufferParser()?.AllBuffers);
+
+            foreach (var file in oldEntries) {
+                await oldAnalyzer.UnloadFileAsync(file);
+            }
+
+            var oldBulkEntries = oldEntries
+                .Where(e => !e.IsTemporaryFile && !e.SuppressErrorList)
+                .Select(e => e.Path)
+                .ToArray();
+
+            var entries = (await AnalyzeFileAsync(oldBulkEntries)).ToList();
+            foreach (var e in oldEntries) {
+                if (e.IsTemporaryFile || e.SuppressErrorList) {
+                    entries.Add(await AnalyzeFileAsync(e.Path, null, e.IsTemporaryFile, e.SuppressErrorList));
                 }
+            }
 
-                foreach (var buffer in buffers) {
-                    oldParser.RemoveBuffer(buffer);
+            foreach (var e in entries) {
+                if (e == null) {
+                    continue;
                 }
 
-                var monitoredResult = await MonitorTextBufferAsync(buffers[0]);
-                if (monitoredResult?.AnalysisEntry != null) {
-                    for (int i = 1; i < buffers.Length; i++) {
-                        monitoredResult.AddBuffer(buffers[i]);
+                ITextBuffer[] buffers;
+                if (oldBuffers.TryGetValue(e.Path, out buffers)) {
```

Comment:
>TryGetValue [](start = 31, length = 11)

We could start using the inline out variable declaration like this:
if (oldBuffers.TryGetValue(e.Path, out ITextBuffer[] buffers))

Reply:
Sometimes I don't mind that, other times it's weird (mostly because the scope of the declaration is _outside_ the if block, not restricted to inside it, so you can reuse the variable later on - not a fan of that).